### PR TITLE
Buffs the valentines event

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Valentines day will now randomly pair up crew members on dates. The paired crewmembers will get an objective to protect each other at all costs.
/:cl:

The valentines holiday event will pair up crew members, giving them objectives to protect each other above all other loyalties.

If there is an odd one out with no valentine they get the martyr objective

I think randomly assigning crew to each other like this will lead to some memorable stories, especially if the nuke op is paired with the captain or something. It will be silly, but will only last a couple days a year.

### CODE REVIEW

**Comment by Shapsy** [comment]
Cute!

**Comment by ShadowDeath6** [comment]
Make sure you can't force redtext by giving a traitor with a kill objective a protect objective on the same person

**Comment by KorPhaeron** [comment]
@ShadowDeath6 

Falling in love with your assassination target is classic /tg/ culture

**Comment by PKPenguin321** [comment]
I've definitely used that tactic to kill a lizard assassination target when ERP was legal

**Comment by kevinz000** [comment]
This does not respect gender
Kek.

**Comment by optimumtact** [comment]
\>currentyear + 1555

\>respecting gender

**Comment by ShadowDeath6** [comment]
>current year being 2557 
>implying the current year is 3572

**Comment by anconfuzedrock** [comment]
tfw somebody undeniably has confirmed that oranges was wrong once

**Comment by JJRcop** [comment]
how about a cuck version where two people get assigned the same date but that date is only assigned one of them

**Comment by jughu** [comment]
70% of this valentine be gay

**Comment by JJRcop** [comment]
KorPhaeron/kortg#28

**Comment by optimumtact** [comment]
I will close the pr if any mention of the word cuck enters it's code

**Comment by KingPhilipIII** [comment]
Captain+Nukie
Clearly the result of faking info on your online dating profile. 

And you must be really desperate if this supersedes all loyalties. 

What do I do if my assassination target is my date?

**Comment by PKPenguin321** [comment]
Anything for love

**Comment by KorPhaeron** [comment]
@KingPhilipIII 

https://1d4chan.org/wiki/Story:Love_Can_Bloom

**Comment by KorPhaeron** [comment]
For some reason people are getting multiple dates, and sometimes getting a date + martyr

**Comment by KorPhaeron** [comment]
@tgstation/commit-access 

Can anyone who can actually code see what might be causing that?

**Comment by Cyberboss** [comment]
Prolly bc you're fucking with the list while in the loop

File: code/modules/events/holiday/vday.dm [hunk 1]
```diff
@@ -19,6 +19,47 @@
...
+
+	var/list/valentines = list()
+	for(var/mob/living/M in player_list)
+		if(!M.stat && M.client && M.mind)
+			valentines |= M
+
+
+	for(var/mob/living/L in valentines)
+		valentines -= L
+		if(valentines.len)
```
Comment (Cyberboss) [inline code/modules/events/holiday/vday.dm]: Try this
```
while(valentines.len)
    var/mob/living/L = pick(valentines)
    valentines -= L
```
etc...

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/events/holiday/vday.dm (hunk 1 above)
Comment (lzimann) [inline code/modules/events/holiday/vday.dm]: pick_n_take will do the removal for you

**Comment by KingPhilipIII** [comment]
@KorPhaeron

How dare you link such heresy.

**Comment by JJRcop** [comment]
I guess I'll just PR the thing if this gets merged either way

**Comment by Robustin** [comment]
>Toss hardsuit and gloves to my crush
>Release Tesla

AT

ALL

COSTS

**Comment by Anonmare** [comment]
>Be Syndicate Elite Nucl-
>Nah I'm just playing ya
>Be John Intern, Syndicate coffee boy turned Nuclear Operative
>Get assigned an objective to detonate a nuclear device stolen from NT on one of Nanotrasen's top secret research facilities
>Lol this gon b ez
>Blast our way onto the station, gunning down security officers, assistants, clowns etc.
>See Captain
>Notice she has the most biggest pair of... Eyes you've ever seen
>Seriously, did this station research anime?
>Gazes locked, you feel a soulbound connection and know you cannot harm your true love
>Turn your sights on your fellow Operatives and gun them down 
>Embrace your beloved
>Renounce your ties to the Syndicate, damn the consequences
>Ride with them to CentCom on the Emergency Shuttle
>Take off your hardsuit, mask and Night Vision Goggles
>Close your eyes and lean in for a kiss...

>Get tased and lased by the jealous HoS, setting off your macrobomb implant and destroying the shuttle
>No survivors
>Love_Can't_Bloom.JPEG

**Comment by KingPhilipIII** [comment]
Hm. I can either fulfill my contract with the company threatening to murder me if I don't comply.
Or I might get laid if the date goes well.

Obvious choice.

**Comment by RandomMarine** [comment]
Or just desperately plea with the other ops to plan to capture the captain alive.

**Comment by calzilla1** [comment]
ERP should be unbanned, but only for the couple

**Comment by KorPhaeron** [comment]
@Cyberboss 

Thanks

**Comment by Robustin** [comment]
Plz remove postround spam.

**Comment by KorPhaeron** [comment]
@Robustin 

Too much work for something that will only be seen three days a year

**Comment by TehZombehz** [comment]
>Get paired with meme lords
Thanks for giving me more reasons to suicide in a round.

**Pull request merged**